subsys/net/l2/ethernet/ethernet.c

```diff
@@ -667,21 +667,6 @@ static int ethernet_send(struct net_if *iface, struct net_pkt *pkt)
 			net_pkt_lladdr_src(pkt)->len =
 						sizeof(struct net_eth_addr);
 			ptype = dst_addr->sll_protocol;
-		} else if (context && net_context_get_type(context) == SOCK_RAW &&
-			   net_context_get_proto(context) == IPPROTO_RAW) {
-			char type = (NET_IPV6_HDR(pkt)->vtc & 0xf0);
-
-			switch (type) {
-			case 0x60:
-				ptype = htons(NET_ETH_PTYPE_IPV6);
-				break;
-			case 0x40:
-				ptype = htons(NET_ETH_PTYPE_IP);
-				break;
-			default:
-				ret = -ENOTSUP;
-				goto error;
-			}
```

Comment:
The ptype variable is now not set for the raw sockets and that variable is given to `ethernet_fill_header()`, so to me it does not look like expected behavior.

Reply:
The idea was that a pkt family (ipv4 or ipv6) is set already in net_context level for AF_PACKET/SOCK_RAW/IPPROTO_RAW. Then later here in ethernet level when sending, an already existing code branches above are handling the ptype correctly for IPv4 and IPv6, if IS_ENABLED(CONFIG_NET_IPV4/IPV6).
This made code simpler. Otherwise, there would be a need for a duplicating similar logic from IPv4 to packet raw, see initial code in this PR.